shell/shell.js

```diff
@@ -16,6 +16,177 @@
+    pause: 'Pause',
+    play: 'Play',
+    mediafastforward: 'MediaFastForward',
+    mediapause: 'MediaPause',
+    mediaplay: 'MediaPlay',
+    mediaplaypause: 'MediaPlayPause',
+    mediarecord: 'MediaRecord',
+    mediarewind: 'MediaRewind',
+    mediatracknext: 'MediaTrackNext',
+    mediatrackprevious: 'MediaTrackPrevious',
+    new: 'New',
+    open: 'Open',
+    print: 'Print',
+    save: 'Save',
+    spellcheck: 'SpellCheck',
+    props: 'Props',
+    select: 'Select',
+    volumeup: 'VolumeUp',
+    volumedown: 'VolumeDown',
+    volumemute: 'VolumeMute',
+    zoomin: 'ZoomIn',
+    zoomout: 'ZoomOut',
+  };
+  for (var idx = 1; idx < 25; idx++) {
+    this.keyModifierAliases[`f${idx}`] = `F${idx}`;
+  }
+  Object.assign(this.keyModifierAliases, keyModifierAliases);
+}
+
+// This is a barebones implementation of matching key-bindings
+// (or as Electron calls them, "key accelerators"). For a more thorough
+// implementation, the Mousetrap library could be used:
+// https://github.com/ccampbell/mousetrap/blob/master/mousetrap.js
+KeyAccelerators.prototype.matches = function(keyEvent, accelerators) {
+  const isMac = this.OS === 'Darwin';
+  const keyModifierAliases = this.keyModifierAliases;
+
+  // Refer to this list for platform-specific behavior for modifier keys:
+  // https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/getModifierState
+  const modifiersToMatch = {};
+
+  // Refer to this list of all possible keys:
+  // https://www.w3.org/TR/DOM-Level-3-Events-key/#key-value-tables
+  const keysToMatch = {};
+
+  (accelerators || '').replace(/\s/g, '').split('+').forEach((accelerator) => {
+    let acceleratorLower = accelerator.toLowerCase();
+    if (acceleratorLower.length === 1) {
+      keysToMatch[acceleratorLower] = null;
+      return;
+    }
+    // Interpret `alt/option` key on Mac as a `meta` key always.
+    if (acceleratorLower.includes('control') || acceleratorLower.includes('ctrl') ||
+        acceleratorLower.includes('command') || acceleratorLower.includes('cmd')) {
+      modifiersToMatch[isMac ? 'Meta' : 'Control'] = null;
+      return;
+    }
+    if ((!isMac && acceleratorLower === 'altgr') ||
+        (isMac && acceleratorLower === 'option')) {
+      modifiersToMatch.AltGr = null;
+      return;
+    }
+    if ((!isMac && acceleratorLower === 'alt') ||
+        (isMac && acceleratorLower === 'option')) {
+      modifiersToMatch.Alt = null;
+      return;
+    }
+    if (isMac && acceleratorLower === 'f5') {
+      modifiersToMatch.Meta = null;
+      // TODO: Figure out why Firefox's `KeyboardEvent#key` is returning `^` instead of `i`.
+      // keysToMatch.r = null;
+      keysToMatch['^'] = null;
+      return;
+    }
+    if (isMac && acceleratorLower === 'f11') {
+      modifiersToMatch.Control = null;
+      modifiersToMatch.Meta = null;
+      keysToMatch.F11 = null;
+      return;
+    }
+    if (acceleratorLower[0] === 'f' && accelerator.length > 1 && accelerator.length < 2) {
```

Comment:
`accelerator.length > 1 && accelerator.length < 2` seems like it can never be true, so this conditional block will never be executed.